src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -2110,20 +2110,49 @@ HICON CreateIconFromRaster(JNIEnv* env, jintArray iconRaster, jint w, jint h)
 void AwtWindow::SetIconData(JNIEnv* env, jintArray iconRaster, jint w, jint h,
                              jintArray smallIconRaster, jint smw, jint smh)
 {
+    HICON hNewIcon = NULL;
+    HICON hNewIconSm = NULL;
+
+    try {
+        hNewIcon = CreateIconFromRaster(env, iconRaster, w, h);
+        if (env->ExceptionCheck()) {
+            if (hNewIcon != NULL) {
+                DestroyIcon(hNewIcon);
+            }
+            return;
+        }
+
+        hNewIconSm = CreateIconFromRaster(env, smallIconRaster, smw, smh);
+        if (env->ExceptionCheck()) {
+            if (hNewIcon != NULL) {
+                DestroyIcon(hNewIcon);
+            }
+            if (hNewIconSm != NULL) {
+                DestroyIcon(hNewIconSm);
+            }
+            return;
+        }
+    } catch (...) {
+        if (hNewIcon != NULL) {
+            DestroyIcon(hNewIcon);
+        }
+        if (hNewIconSm != NULL) {
+            DestroyIcon(hNewIconSm);
+        }
+        return;
+    }
+
     HICON hOldIcon = NULL;
     HICON hOldIconSm = NULL;
-    //Destroy previous icon if it isn't inherited
     if ((m_hIcon != NULL) && !m_iconInherited) {
```

Comment:
I am not sure why the check for NULL is needed .. but it is OK.

Reply:
What do you mean? That it could've been just `if ((m_hIcon) && !m_iconInherited)`?

This line is unchanged.  
In the majority of cases the surrounding code uses explicit `!= NULL` in `if`-statements.